Vostok.ClusterClient.Topology.SD/Transforms/KeepLastNonEmptyTopologyTransform.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using Vostok.Logging.Abstractions;
+using Vostok.ServiceDiscovery.Abstractions;
+
+namespace Vostok.Clusterclient.Topology.SD.Transforms
+{
+    [PublicAPI]
+    public class KeepLastNonEmptyTopologyTransform : IServiceTopologyTransform
+    {
+        private readonly ILog log;
+        private volatile IReadOnlyList<Uri> lastSeenReplicas;
+
+        public KeepLastNonEmptyTopologyTransform(ILog log)
+        {
+            this.log = log ?? LogProvider.Get();
+        }
+
+        public IEnumerable<Uri> Transform(IServiceTopology topology)
+        {
+            var replicas = topology.Replicas;
+            if (replicas.Count == 0 && lastSeenReplicas?.Count > 0)
+            {
+                log.Warn("New observed topology is empty, but we have a cached one. Last seen cached topology will be used.");
```

Comment:
These log messages are going to be spammy (one for each request while there are no replicas in current topology).